paddle/optimizer/adadelta_optimizer.h

```diff
@@ -0,0 +1,39 @@
+#ifndef PADDLE_ADADELTA_OPTIMIZER_H_
+#define PADDLE_ADADELTA_OPTIMIZER_H_
+
+#include "parameter_optimizer.h"
+
+namespace paddle {
+namespace optimizer {
+
+class AdadeltaOptimizer : public ParameterOptimizer {
+public:
+  using ParameterOptimizer::parameter_;
```

Comment:
We should not use public variable if it's not intended for public. It breaks encapsulation. Protected is fine for me. I understand being consistent is important, but we should not be consistent with poor design, otherwise how can we improve code quality?